spyder/app/mainwindow.py

```diff
@@ -3203,35 +3505,66 @@ def set_prefs_size(self, size):
 
     #---- Shortcuts
     def register_shortcut(self, qaction_or_qshortcut, context, name,
-                          add_shortcut_to_tip=False, plugin_name=None):
+                          add_shortcut_to_tip=True, plugin_name=None):
         """
         Register QAction or QShortcut to Spyder main application,
         with shortcut (context, name, default)
         """
         self.shortcut_data.append((qaction_or_qshortcut, context,
                                    name, add_shortcut_to_tip, plugin_name))
 
+    def unregister_shortcut(self, qaction_or_qshortcut, context, name,
+                            add_shortcut_to_tip=True, plugin_name=None):
+        """
+        Unregister QAction or QShortcut from Spyder main application.
+        """
+        data = (qaction_or_qshortcut, context, name, add_shortcut_to_tip,
+                plugin_name)
+
+        if data in self.shortcut_data:
+            self.shortcut_data.remove(data)
+
     def apply_shortcuts(self):
-        """Apply shortcuts settings to all widgets/plugins"""
+        """Apply shortcuts settings to all widgets/plugins."""
         toberemoved = []
+        # TODO: Check shortcut existence based on action existence, so that we
+        # can update shortcut names without showing the old ones on the
+        # preferences
         for index, (qobject, context, name, add_shortcut_to_tip,
                     plugin_name) in enumerate(self.shortcut_data):
-            keyseq = QKeySequence(CONF.get_shortcut(context, name,
-                                                    plugin_name))
             try:
-                if isinstance(qobject, QAction):
-                    if sys.platform == 'darwin' and \
-                      qobject._shown_shortcut == 'missing':
-                        qobject._shown_shortcut = keyseq
-                    else:

```

Comment:
Why are you removing this logic? This took us years to get it right.

Reply:
It is moved bellow.